addon/components/polaris-resource-list/filter-control.js

```diff
@@ -0,0 +1,328 @@
+    let value = get(appliedFilter, 'value');
+    let label = get(appliedFilter, 'label');
+    if (label) {
+      return label;
+    }
+
+    let filters = this.get('filters') || [];
+
+    let filter = filters.find((filter) => {
+      let minKey = get(filter, 'minKey');
+      let maxKey = get(filter, 'maxKey');
+      let operatorText = get(filter, 'operatorText');
+
+      if (minKey || maxKey) {
+        return get(filter, 'key') === key || minKey === key || maxKey === key;
+      }
+
+      if (operatorText && typeof operatorText !== 'string') {
+        return (
+          get(filter, 'key') === key ||
+          operatorText.filter((operator) => get(operator, 'key') === key)
+            .length === 1
+        );
+      }
+
+      return get(filter, 'key') === key;
+    });
+
+    if (!filter) {
+      return value;
+    }
+
+    let filterOperatorLabel = findOperatorLabel(filter, appliedFilter);
+    let filterLabelByType = this.findFilterLabelByType(filter, appliedFilter);
+
+    if (!filterOperatorLabel) {
+      return `${filter.label} ${filterLabelByType}`;
+    }
+
+    return `${filter.label} ${filterOperatorLabel} ${filterLabelByType}`;
+  },
+
+  findFilterLabelByType(filter, appliedFilter) {
+    let appliedFilterValue = get(appliedFilter, 'value');
+
+    if (get(filter, 'type') === FilterType.Select) {
+      let foundFilterOption = get(filter, 'options').find(
+        (option) =>
+          typeof option === 'string'
+            ? option === appliedFilterValue
+            : get(option, 'value') === appliedFilterValue
+      );
+
+      if (foundFilterOption) {
+        return typeof foundFilterOption === 'string'
+          ? foundFilterOption
+          : get(foundFilterOption, 'label');
+      }
+    }
+
+    if (get(filter, 'type') === FilterType.DateSelector) {
+      if (get(filter, 'key') === get(appliedFilter, 'key')) {
+        return getFilterLabelForValue(get(appliedFilter, 'value'));
+      }
+
+      if (get(appliedFilter, 'key') === get(filter, 'maxKey')) {
+        return `before ${formatDateForLabelDisplay(
+          get(appliedFilter, 'value')
+        )}`;
+      }
+
+      if (get(appliedFilter, 'key') === get(filter, 'minKey')) {
+        return `after ${formatDateForLabelDisplay(
+          get(appliedFilter, 'value')
+        )}`;
+      }
+    }
+
+    return appliedFilterValue;
+  },
+});
+
+function idFromFilter(appliedFilter) {
+  return `${get(appliedFilter, 'key')}-${get(appliedFilter, 'value')}`;
+}
+
+function formatDateForLabelDisplay(date) {
+  if (isNaN(new Date(date).getTime())) {
+    return date;
+  }
+
+  return new Date(date.replace(/-/g, '/')).toLocaleDateString();
+}
+
+function findOperatorLabel(filter, appliedFilter) {
+  let operatorText = get(filter, 'operatorText');
+
+  if (
+    filter.type === FilterType.DateSelector &&
+    (get(appliedFilter, 'key') === get(filter, 'minKey') ||
+      get(appliedFilter, 'key') === get(filter, 'maxKey'))
```

Comment:
It seems you have also been bitten by the weird `||` indentation bug 😝

Reply:
But it's _prettier_ this way :trollface: